noxfile.py

```diff
@@ -479,10 +487,7 @@ def check_changelog(session: nox.Session):
 def set_minimal_package_versions(session: nox.Session):
     from collections import defaultdict
 
-    try:
-        import tomllib as toml
-    except ImportError:
-        import toml
+    assert toml is not None, "requires Python 3.11 or toml package"
```

Comment:
Isn't `assert` easy to become a no-op for example in compiled byte code and hence should not be used for user facing errors. (Admittedly, the noxfile is not particularly user facing, but personally I would rather use this as an excuse to just require TOML support unconditionally, i.e. raise this message directly in line 22.

Reply:
I agree this is a bit clumsy. My reason to not want unconditional TOML support is that we'll end up having to adjust a lot of CI jobs to download the package needlessly.

Would you accept a compromise if I changed the assert to use proper `nox` APIs to fail the job?

Reply:
> Would you accept a compromise if I changed the assert to use proper nox APIs to fail the job?

Yes, that would be fine by me.

Reply:
Done 👍